### PR TITLE
Convert GOR4 to run under cffi instead of Cython

### LINKED ISSUE #333: Convert GOR IV to use cffi instead of Cython
Then we can run the code under pypy.

### PR DESCRIPTION
**Pull request by terrycojones**
This is based on #332, which should be reviewed & merged first.

Made GOR4 run under cffi instead of Cython.

Also: Added (not very good) install doc for OS X. Added a median function (and tests) that works under python 2, 3, and pypy. There are 2 test errors running under pypy which I think we should fix in another branch (they are due to virtualenv creating a non-framework version of Python, which causes matplotlib to fail on import).

Fixes #333.

### CODE REVIEW

File: dark/gor4.py [hunk 1]
```diff
@@ -17,32 +17,25 @@
...
 
-        self._state = cgor4.initialize(c_sequenceFile, c_secondaryFile,
-                                       &error)
-        if error:
-            raise Exception('Error in gor4-base.c initialize function.')
+        error = ffi.new('int *')
+        state = lib.initialize(ffi.new('char[]', sequenceFile),
+                               ffi.new('char[]', secondaryFile), error)
+        if state == 0:
+            raise Exception('Error in gor4 initialization.')
```
Comment (bamueh) [inline dark/gor4.py]: Is it possible to make this more specific? I guess not?

File: dark/gor4.py [hunk 2]
```diff
@@ -17,32 +17,25 @@
...
-        """
-        Free the storage associated with self._state.
-        """
-        if self._state is not NULL:
-            cgor4.finalize(self._state)
-
-    def __init__(self, sequenceFile=_SEQUENCES, secondaryFile=_SECONDARY):
-        pass
+        # ffi.gc returns a copy of the cdata object which will have the
+        # destructor (in this case ``finalize``) called when the Python
```
Comment (bamueh) [inline dark/gor4.py]: Why do you have the characters around finalize? Can it just be 'finalize'?

Comment (terrycojones) [inline dark/gor4.py]: That was copy / pasted from an example online. Fixed.

File: dark/gor4.py (hunk 1 above)
Comment (terrycojones) [inline dark/gor4.py]: Well, this was actually a bug. It should have been checking the `error` value, not `state`.

There doesn't need to be anything more specific here, as the GOR4 C code will print an error message to stderr if something goes wrong.  I just tried some non-existent filenames for the initialization to make sure that happens.

**Pull request merged**